File: sboot-atom-usuarios/src/main/java/br/com/books/web/usuarios/service/UsuarioService.java
package br.com.books.web.usuarios.service;

import br.com.books.web.usuarios.dto.UsuarioRequestDTO;
import br.com.books.web.usuarios.exceptions.UsuarioNaoEncontradoException;
import br.com.books.web.usuarios.model.Usuario;

import java.util.List;

public interface UsuarioService {

    List<Usuario> obterTodosUsuarios();

    Usuario obterUsuarioPorId(Long id) throws UsuarioNaoEncontradoException;

    Usuario adicionarUsuario(UsuarioRequestDTO usuario);

    Usuario atualizarUsuario(Long id, UsuarioRequestDTO detalhesUsuario) throws UsuarioNaoEncontradoException;

    void deletarUsuario(Long id) throws UsuarioNaoEncontradoException;
}

File: front-books-on-the-web/src/app/services/usuario.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface UsuarioRequestDTO {
  nome: string;
  email: string;
  telefone: string;
}

export interface UsuarioResponseDTO {
  idUsuario: number;
  nome: string;
  email: string;
  telefone: string;
}

@Injectable({
  providedIn: 'root',
})
export class UsuarioService {
  private apiUrl = 'http://localhost:8086/api/usuarios';

  constructor(private http: HttpClient) {}

  obterTodosUsuarios(): Observable<UsuarioResponseDTO[]> {
    return this.http.get<UsuarioResponseDTO[]>(this.apiUrl);
  }

  obterUsuarioPorId(id: number): Observable<UsuarioResponseDTO> {
    return this.http.get<UsuarioResponseDTO>(`${this.apiUrl}/${id}`);
  }

  adicionarUsuario(usuario: UsuarioRequestDTO): Observable<UsuarioResponseDTO> {
    return this.http.post<UsuarioResponseDTO>(this.apiUrl, usuario);
  }

  atualizarUsuario(id: number, usuario: UsuarioRequestDTO): Observable<UsuarioResponseDTO> {
    return this.http.put<UsuarioResponseDTO>(`${this.apiUrl}/${id}`, usuario);
  }

  deletarUsuario(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
